Sources/AEPOptimize/Optimize.swift

```diff
@@ -241,20 +241,38 @@ public class Optimize: NSObject, Extension {
         // add the Edge event to update propositions in the events queue.
         eventsQueue.add(edgeEvent)
 
-        // Increase timeout to 5s to ensure edge requests have enough time to complete.
-        MobileCore.dispatch(event: edgeEvent, timeout: 5) { responseEvent in
+        // Increase timeout to 10s to ensure edge requests have enough time to complete.
+        MobileCore.dispatch(event: edgeEvent, timeout: 10) { responseEvent in
             guard
                 let responseEvent = responseEvent,
                 let requestEventId = responseEvent.requestEventId
             else {
-                // response event failed or timed out, remove this event's ID from the requested event IDs dictionary and kick-off queue.
+                // response event failed or timed out, remove this event's ID from the requested event IDs dictionary, dispatch an error response event and kick-off queue.
                 self.updateRequestEventIdsInProgress.removeValue(forKey: edgeEvent.id.uuidString)
                 self.propositionsInProgress.removeAll()
-
+                let timeoutError = AEPOptimizeError(
+                    type: nil,
+                    status: 408,
+                    title: "Request Timeout",
+                    detail: "Update proposition request resulted in a timeout.",
+                    aepError: AEPError.callbackTimeout
+                )
+                self.dispatch(event: event.createErrorResponseEvent(timeoutError))
                 self.eventsQueue.start()
                 return
             }
 
+            // response event to provide success callback to updateProposition public api
+            let responseEventToSend = event.createResponseEvent(
```

Comment:
Considering we're also using this same event type and source in `processGetPropositions` do you think it makes sense to move this also to Event+Optimize, similar to `createErrorResponseEvent`?

Reply:
Yes, since now we are returning ([DecisionScope: OptimizeProposition]?, AEPOptimizeError?) in update proposition's api, we can create a common util function in Event+Optimize.swift. However, in update proposition callback we can have scenario where we have successful [DecisionScope: OptimizeProposition] and AEPOptimizeError both if one upstream gives error and other succeed. So I tried creating a common function which is like this -
**createCallbackResponseEvent(_ data: [DecisionScope: OptimizeProposition], _ error: AEPOptimizeError?) -> Event** and in case of get proposition i tried sending error as nil which failed some of the test cases. I am debugging that and once I am clear on that I will push it or will include this in next PR which will cover other edge error handling apart from timeout.